### ISSUE TITLE
Block time is wrong

### ISSUE THREAD

**Issue by chevdor** [ethcore/parity#3888]
Since #3170 was not reopened, here is a new issue.
Some details are in #3170.

Both screenshots were taken today 18.12.2016 around 10:30 AM.

This shows the CURRENT time for any block and not the block time. As a matter of fact, if syncs halts, the block number remains the same but the time keep on updating, showing the current time apparently.
<img width="183" alt="screen shot 2016-12-18 at 10 27 16" src="https://cloud.githubusercontent.com/assets/738724/21292662/fa23d37c-c50c-11e6-9511-d8b1686ab2c9.png">

We see here that the blocktime is far from being 10:25 AM
<img width="525" alt="screen shot 2016-12-18 at 10 27 23" src="https://cloud.githubusercontent.com/assets/738724/21292663/fa26fa7a-c50c-11e6-81c5-9122a2eb58ea.png">

**Comment by jacogr** [comment]
Been monitoring it with some additional debug info -

1. Timestamp displayed is not current, rather the blocktime (also not when it is received, it is the time as marked)
2. Have not been able to see tomorrow/etc. rather time only changed on new block

In short, not sure what is up here.

**Comment by derhuerst** [comment]
@chevdor Have you come across this issue again after #3894 got merged?

**Comment by chevdor** [comment]
I just rebuilt and tested and it looks fine now. Thank you both.
